<launch>

    <!-- <group ns="xiroi"> -->

      <!-- MEMSENSE IMU NODE -->
      <node pkg="memsense_imu" type="imu_node" name="imu_node" output="screen">
        <param name="polling_rate" value="150.00" type="double" />
        <param name="filter_rate"  value="10.0" type="double" />
        <param name="frame_id"     value="imu" type="string" />
        <param name="serial_port"  value="/dev/serial/by-id/usb-Silicon_Labs_CP2101_USB_to_UART_Bridge_Controller_0001-if00-port0" type="string"/>
        <param name="imu_type"     value="nIMU_3temp" type="string" />

        <param name="accels_range" value="19.620000000000001" type="double" />
        <param name="accel_bias_x" value="0.02" type="double" />
        <param name="accel_bias_y" value="0.0" type="double" />
        <param name="accel_bias_z" value="0.0" type="double" />
        <param name="accels_var"   value="0.0" type="double" />

        <param name="gyros_range"  value="2.617993877991494" type="double" />
        <param name="gyro_bias_x"  value=" 0.009295" type="double" />
        <param name="gyro_bias_y"  value=" 0.017145" type="double" />
        <param name="gyro_bias_z"  value="-0.002915" type="double" />
        <param name="gyros_var"    value="0.0" type="double" />

        <param name="mag_bias_x"   value="0.0" type="double" />
        <param name="mag_bias_y"   value="0.0" type="double" />
        <param name="mag_bias_z"   value="0.0" type="double" />
        <param name="mags_range"   value="1.8999999999999999" type="double" />
        <param name="mags_var"     value="0.0" type="double" />

        <remap from="imu/data" to="sensors/imu_raw_driver"/>
        <remap from="imu/data_calibrated" to="sensors/imu_raw_calibrated"/>
        <remap from="imu/data_filtered" to="sensors/imu_raw_filtered"/>
        <remap from="imu/data_filtered_calibrated" to="sensors/imu_raw_filtered_calibrated"/>
        <remap from="imu/mag" to="sensors/imu_mag"/>
        <remap from="imu/mag_calibrated" to="sensors/imu_mag_calibrated"/>
        <remap from="imu/mag_filtered" to="sensors/imu_mag_filtered"/>
        <remap from="imu/mag_filtered_calibrated" to="sensors/imu_mag_filtered_calibrated"/>
      </node>

      <!-- IMU FILTER MADGWICK NODE -->
      <node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter_node" output="screen">
        <param name="stateless" value="false"/>
        <param name="fixed_frame" value="odom"/>
        <param name="publish_tf" value="false"/>
        <param name="constant_dt" value="0.0"/>
        <param name="publish_debug_topics" value="false"/>
        <param name="use_mag" value="true"/>
        <param name="use_magnetic_field_msg" value="true"/>
        <param name="world_frame" value="ned"/>

        <remap from="imu/data_raw" to="sensors/imu_raw_driver"/>     <!-- subscriber -->
        <remap from="imu/mag" to="sensors/imu_mag"/>          <!-- subscriber -->
        <remap from="imu/data" to="sensors/imu_raw"/>        <!-- publisher -->
      </node>

    <!-- </group> -->
</launch>